### PR TITLE
Update reftest graph when node changes to support

### PR DESCRIPTION
**Pull request by Cactusmachete**
Modified conditional in manifest.py to check for a reftest node changing
to another type.
Wrote a test in test_manifest.py for the same, but this needs some review.

Closes https://github.com/w3c/web-platform-tests/issues/9671

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/9732)

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-03-01 12:11:15
Finished: 2018-03-01 12:28:19





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25093)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/347733830)

File: tools/manifest/tests/test_manifest.py [hunk 1]
```diff
@@ -254,6 +254,31 @@ def test_reftest_computation_chain_update_test_type():
...
+    assert list(m) == [("reftest", test1.path, {test1}),
+                       ("reftest_node", test2.path, {test2})]
+
+    #test2 changes to support type
+    s3 = SourceFileWithTest("test2", "1"*40, item.SupportFile)
+
+    # This doesn't check whether the reftest graph was computed or not,
+    # the m.update([s3]) assert will just check whether m.update() returned True (I think).
+    # Regardless, the test works even when the graph isn't being recomputed.
+    assert m.update([s3]) is True
```
Comment (jgraham) [inline tools/manifest/tests/test_manifest.py]: Then here I think you want to call `m.update([s1, s2])` again.

File: tools/manifest/tests/test_manifest.py [hunk 2]
```diff
@@ -254,6 +254,31 @@ def test_reftest_computation_chain_update_test_type():
...
+    assert m.update([s1, s2]) is True
+
+    test1 = s1.manifest_items()[1][0]
+    test2 = s2.manifest_items()[1][0]
+
+    assert list(m) == [("reftest", test1.path, {test1}),
+                       ("reftest_node", test2.path, {test2})]
+
+    #test2 changes to support type
+    s3 = SourceFileWithTest("test2", "1"*40, item.SupportFile)
```
Comment (jgraham) [inline tools/manifest/tests/test_manifest.py]: So I think you want to redefine `s2` here rather than creating a new `s3`

**Review by jgraham — CHANGES_REQUESTED** [review]
I think the way you've written the test effetively creates an entirely new manifest that's not related to the previous one.

Note that in general one should push additional commits to the existing PR (by pushing to the same branch) rather than creating a new PR with changes.

File: tools/manifest/tests/test_manifest.py (hunk 2 above)
Comment (Cactusmachete) [reply tools/manifest/tests/test_manifest.py]: I made the changes and it works as it should now!
Still need to understand why, though. 
Thanks a lot :smile: 

And I'm sorry I made a new PR for this, I'll ensure I push to the same branch from now on.

**Review by jgraham — APPROVED** [review]
Thanks!

**Pull request merged**